var/spack/repos/builtin/packages/pmix/package.py

```diff
@@ -49,11 +52,34 @@ class Pmix(AutotoolsPackage):
 
     depends_on('libevent@2.0.20:2.0.22,2.1.8')
     depends_on('hwloc@1.11.0:1.11.99,2.0.1:', when='@3.0.0:')
+    depends_on("m4", type=("build"), when="@master")
+    depends_on("autoconf", type=("build"), when="@master")
+    depends_on("automake", type=("build"), when="@master")
+    depends_on("libtool", type=("build"), when="@master")
+    depends_on("perl", type=("build"), when="@master")
+
+    def autoreconf(self, spec, prefix):
+        """Only needed when building from git checkout"""
+        # If configure exists nothing needs to be done
+        if os.path.exists(self.configure_abs_path):
+            return
+        # Else bootstrap with autotools
+        perl = which('perl')
+        perl('./autogen.pl')
 
     variant('pmi_backwards_compatibility',
             default=True,
             description="Toggle pmi backwards compatibility")
 
+    variant('restful',
+            default=False,
+            description="allow a PMIx server to request services from "
+            "a system-level REST server")
+    depends_on('curl', when="+restful")
+    depends_on('jansson@2.11:', when="+restful")
+
+    conflicts('@:3.9.9', when='+restful')
```

Comment:
This isn't required, but I usually like to sort things in packages like:

1. Docstring
2. URLs
3. Versions
4. Variants
5. Dependencies
6. Conflicts
7. Functions

I would definitely move autoreconf down above configure_args, and probably move the variant between versions and dependencies.

Reply:
thanks for the guidance.  reordered things in a third commit